packages/iris-grid/src/mousehandlers/IrisGridTokenMouseHandler.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  EventHandlerResult,
+  Grid,
+  GridMouseHandler,
+  GridPoint,
+  GridUtils,
+  isLinkToken,
+  TokenBox,
+} from '@deephaven/grid';
+import deepEqual from 'deep-equal';
+import IrisGrid from '../IrisGrid';
+
+class IrisGridTokenMouseHandler extends GridMouseHandler {
+  private irisGrid: IrisGrid;
+
+  // Stores the current hovered token box if it exists with translated coordinates
+  private currentLinkBox?: TokenBox;
+
+  constructor(irisGrid: IrisGrid) {
+    super();
+
+    this.irisGrid = irisGrid;
+  }
+
+  private destroyTooltip(): void {
+    this.irisGrid.setState({ linkHoverTooltipProps: null });
+  }
+
+  isHoveringLink(gridPoint: GridPoint, grid: Grid): boolean {
+    const { column, row, x, y } = gridPoint;
+    const { renderer, metrics } = grid;
+
+    if (column == null || row == null || metrics == null) {
+      this.currentLinkBox = undefined;
+      return false;
+    }
+
+    if (this.currentLinkBox != null) {
+      const { x1: left, y1: top, x2: right, y2: bottom } = this.currentLinkBox;
+      if (x >= left && x <= right && y >= top && y <= bottom) {
+        return true;
+      }
+    }
+
+    const renderState = grid.updateRenderState();
+    const tokensInCell = renderer.getTokenBoxesForVisibleCell(
+      column,
+      row,
+      renderState
+    );
+
+    // Loop through each link and check if cursor is in bounds
+    for (let i = 0; i < tokensInCell.length; i += 1) {
+      const translatedTokenBox = GridUtils.translateTokenBox(
+        tokensInCell[i],
+        metrics
+      );
+      const { x1: left, x2: right, y1: top, y2: bottom } = translatedTokenBox;
+      if (
+        x >= left &&
+        x <= right &&
+        y >= top &&
+        y <= bottom &&
+        isLinkToken(tokensInCell[i].token)
+      ) {
+        this.currentLinkBox = translatedTokenBox;
+        return true;
+      }
```

Comment:
Minor optimization - you could check if it's a link token before translating the token:
```
      if (isLinkToken(tokensInCell[i].token)) {
      
      const translatedTokenBox = GridUtils.translateTokenBox(
        tokensInCell[i],
        metrics
      );
      const { x1: left, x2: right, y1: top, y2: bottom } = translatedTokenBox;
      if (
        x >= left &&
        x <= right &&
        y >= top &&
        y <= bottom
      ) {
        this.currentLinkBox = translatedTokenBox;
        return true;
      }
      }
```